core/src/test/scala/unit/kafka/admin/ReassignPartitionsCommandTest.scala

```diff
@@ -218,8 +202,183 @@ class ReassignPartitionsCommandTest extends ZooKeeperTestHarness with Logging wi
+  def shouldNotOverwriteExistingPropertiesWhenLimitIsAdded(): Unit = {
+    //Given
+    val existing = mutable.Map(TopicAndPartition("topic1", 0) -> Seq(100, 101))
+    val proposed = mutable.Map(TopicAndPartition("topic1", 0) -> Seq(101, 102))
+
+    //Setup
+    val zk = stubZK(existing)
+    val admin = createMock(classOf[AdminUtilities])
+    val propsCapture: Capture[Properties] = newCapture(CaptureType.ALL)
+    val assigner = new ReassignPartitionsCommand(zk, proposed, admin)
+    expect(admin.changeBrokerConfig(is(zk), anyObject().asInstanceOf[List[Int]], capture(propsCapture))).anyTimes()
+
+    //Given there is some existing config
+    expect(admin.fetchEntityConfig(is(zk), is(ConfigType.Broker), anyString())).andReturn(
+      propsWith("useful.key", "useful.value")).atLeastOnce()
+
+    replay(admin)
+
+    //When
+    assigner.maybeLimit(1000)
+
+    //Then other property remains
+    for (actual <- propsCapture.getValues) {
+      assertEquals("useful.value", actual.getProperty("useful.key"))
+      assertEquals("1000", actual.getProperty(DynamicConfig.Broker.LeaderReplicationThrottledRateProp))
+      assertEquals("1000", actual.getProperty(DynamicConfig.Broker.FollowerReplicationThrottledRateProp))
+    }
+    assertEquals(3, propsCapture.getValues.size()) //3 brokers
+  }
+
+  @Test
+  def shouldRemoveThrottleLimitFromAllBrokers(): Unit = {
+    //Given 3 brokers, but with assignment only covering 2 of them
+    val brokers = Seq(100, 101, 102)
+    val proposed = mutable.Map(TopicAndPartition("topic1", 0) -> Seq(100, 101))
+    val status = mutable.Map(TopicAndPartition("topic1", 0) -> ReassignmentCompleted)
+    val existingBrokerConfigs = propsWith(
+      (DynamicConfig.Broker.FollowerReplicationThrottledRateProp, "10"),
+      (DynamicConfig.Broker.LeaderReplicationThrottledRateProp, "100"),
+      ("useful.key", "value")
+    )
+
+    //Setup
+    val zk = stubZK(brokers = brokers)
+    val admin = createMock(classOf[AdminUtilities])
+    val propsCapture: Capture[Properties] = newCapture(CaptureType.ALL)
+    expect(admin.fetchEntityConfig(is(zk), is(ConfigType.Topic), anyString())).andStubReturn(new Properties)
+    expect(admin.changeBrokerConfig(is(zk), anyObject().asInstanceOf[Seq[Int]], capture(propsCapture))).anyTimes()
+    //Stub each invocation as EasyMock caches the return value which can be mutated
+    expect(admin.fetchEntityConfig(is(zk), is(ConfigType.Broker), is("100"))).andReturn(copyOf(existingBrokerConfigs))
+    expect(admin.fetchEntityConfig(is(zk), is(ConfigType.Broker), is("101"))).andReturn(copyOf(existingBrokerConfigs))
+    expect(admin.fetchEntityConfig(is(zk), is(ConfigType.Broker), is("102"))).andReturn(copyOf(existingBrokerConfigs))
+    replay(admin)
+
+    //When
+    ReassignPartitionsCommand.removeThrottle(zk, proposed, status, admin)
+
+    //Then props should have gone (dummy remains)
+    for (capture <- propsCapture.getValues)
+      assertEquals("value", capture.get("useful.key"))
```

Comment:
I think we should check that the properties are null, I broker the code on purpose and it still passed without that.